### PR TITLE
Add support for `Prim.Char`

### PR DESCRIPTION
**Pull request by felixSchl**
Add support for `Prim.Char` and update the tests

### CODE REVIEW

File: test/Main.purs [hunk 1]
```diff
@@ -8,6 +8,7 @@ import Control.Monad.Eff.Console
 import Data.Either
 
 data Foo = Foo Number String | Bar Number | Quux (Array String) | Baz {a :: Maybe String, bq :: Number} String
+         | Corge (Array Char)
```
Comment (paf31) [inline test/Main.purs]: "Corge"? :smile:

**Comment by paf31** [comment]
Thanks! Looks good to me. @gbaz Any comments on this?

File: test/Main.purs (hunk 1 above)
Comment (felixSchl) [inline test/Main.purs]: I just went by wikipedia (https://en.wikipedia.org/wiki/Metasyntactic_variable#English)

Comment (paf31) [inline test/Main.purs]: Ah ok :) Hadn't heard that one before..

Comment (gbaz) [inline test/Main.purs]: looks good to me.

**Comment by gbaz** [comment]
Looks good to me.

**Comment by paf31** [comment]
@garyb Do you know if we added `generics` dependencies anywhere in core?

**Comment by garyb** [comment]
I don't know off the top of my head, and I don't have an up to date version of the script I've been using - I'll check when I get back home (and actually update the repo so other people can use it too).

**Comment by garyb** [comment]
I think `purescript-argonaut-codecs` is the only dependency in core and contrib currently.

**Pull request merged**